webrepl_cli.py

```diff
@@ -191,33 +194,43 @@ def parse_remote(remote):
 
 
 def main():
-
-    if len(sys.argv) != 3:
+    if len(sys.argv) not in [3, 5]:
         help(1)
 
-    if ":" in sys.argv[1] and ":" in sys.argv[2]:
+    if len(sys.argv) == 5 and '-p' in sys.argv:
+        passwd_specifier_i = sys.argv.index('-p')
+        passwd = sys.argv[passwd_specifier_i + 1]
+        arguments = [sys.argv[i] for i in range(len(sys.argv)) 
+                                 if  i not in (passwd_specifier_i, passwd_specifier_i + 1)]
+    else:
+        import getpass
+        passwd = getpass.getpass()
+        arguments = sys.argv
+
+    if ":" in arguments[1] and ":" in arguments[2]:
         error("Operations on 2 remote files are not supported")
-    if ":" not in sys.argv[1] and ":" not in sys.argv[2]:
+    if ":" not in arguments[1] and ":" not in arguments[2]:
         error("One remote file is required")
 
-    if ":" in sys.argv[1]:
+
+    if ":" in arguments[1]:
         op = "get"
-        host, port, src_file = parse_remote(sys.argv[1])
-        dst_file = sys.argv[2]
+        host, port, src_file = parse_remote(arguments[1])
+        dst_file = arguments[2]
         if os.path.isdir(dst_file):
             basename = src_file.rsplit("/", 1)[-1]
             dst_file += "/" + basename
+        infomsg("%s %s from %s:%d (passwd:%s) and save as %s" % 
+                            (op, src_file, host, port, passwd, os.path.abspath(dst_file)))
```

Comment:
Please leave the original messages as they were.  It particular you don't want to print out the password!